### ISSUE TITLE
JEP358NPEMessageTests_0 and JEP358NPEMessageTests-noDebugInfo_0 can't find testng.xml

### ISSUE THREAD

**Issue by sophia-guo** [AdoptOpenJDK/run-aqa#40]
JEP358NPEMessageTests_0 and JEP358NPEMessageTests-noDebugInfo_0 failed as:

```
java.io.FileNotFoundException: /home/runner/work/run-aqa/run-aqa/jvmtest/functional/Java14andUp/testng.xml (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:211)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:153)
	at org.testng.xml.Parser.parse(Parser.java:148)

```
 at https://github.com/AdoptOpenJDK/run-aqa/runs/1376392753?check_suite_focus=true

The reason is Java14andUp has been set only building for `openj9` and `ibm` in build.xml. Hence no tests under this folder should be run by other `impl`.  Test should be limited to 

```
        <impls>
            <impl>openj9</impl>
            <impl>ibm</impl>
        </impls>
```

@llxia please confirm if all tests under Java14andUp is not for other impls. If yes, those tests should be limited to their impls.

**Comment by llxia** [comment]
@JasonFengJ9 are JEP358NPEMessageTests and JEP358NPEMessageTests-noDebugInfo openj9 and ibm specific test?

**Comment by JasonFengJ9** [comment]
@llxia yes, these two tests apply to `OpenJ9` & `IBM`, not other implementations. 

My understanding is that the test needs `<impls>` only if it applies to `openj9` or `ibm` (IBM is here for historical reason), if the test is for both, then there is no `<impls>` required. 
I understand that `AQA` test framework supports broader JVM families, and need to know the implementation restrictions. However it is kind of strange that a test implemented within `OpenJ9` project has to be specific about its own `<impls>`.
I feel the framework should be aware of the source project of a test, and assume its default support implementation, and only apply the test to other `impls` if possible.
On the other words, I think all tests from `OpenJ9` project should be restricted to `OpenJ9` & `IBM` by default, this can avoid boilerplate code for such `<impls>` causes. 

Having said that, I can add such `<impls>` restriction before `AQA` framework can handle such scenarios.

**Comment by JasonFengJ9** [comment]
Created https://github.com/eclipse/openj9/pull/11146.

**Comment by llxia** [comment]
Thanks @JasonFengJ9 for the PR.

The current behavior is that if `<impls>` is not provided, then it defaults to all JVM. 

IMHO, having the AQA/TKG framework to assume the default supported JVM for tests complicates the story. Instead of the test explicitly determines its supported JVM, it adds one more hidden layer to make this decision (in a different repo). It means a map of git repo to `<impls>` needs to be stored somewhere in the framework and users that change multiple repos need to be clear the default `<impls>` value is different from repo to repo.

**Comment by JasonFengJ9** [comment]
@llxia totally understandable, the framework works well if each test provides concrete `<impls>` restrictions or no `<impls>` if the test applies to all JVMs. 
Honestly I didn't have other JVMs in mind when writing those tests, not sure if that should be a requirement. A quick search in `OpenJ9` repo shows that only handful of tests don't have `<impls>`, i.e., default to all JVMs. In another word, ~500 test suites are restricted to `OpenJ9/IBM` as per `<impls>` description however I suspect many of them are not `J9` specific.

Basically my point is that `<impls>` restriction is not necessary accurate restriction, particularly regarding other JVMs. 
Does `AQA` framework have a plan to harness the tests from other open source projects? If so, do we also expect `OpenJ9` style test structure or have a different way to consume those tests?

**Comment by smlambert** [comment]
Understandable that you did not consider the tests to be applicable to other implementations, but in many cases openj9 test material could be applicable (if we take care of openj9 specific JVM_OPTIONS).

AQA harnesses tests from openjdk, and from many many [external](https://github.com/AdoptOpenJDK/openjdk-tests/tree/master/external) (3rd party applications and frameworks) and [perf](https://github.com/AdoptOpenJDK/openjdk-tests/tree/master/perf) benchmarks to test all implementations.  

Just as we assume all openjdk tests can apply to openj9 (not just hotspot), unless otherwise specified, we assume openj9 tests can be applied to hotspot unless otherwise specified.

related: https://github.com/eclipse/openj9/issues/5066 and https://github.com/eclipse/openj9/issues/10227

**Comment by JasonFengJ9** [comment]
> Just as we assume all openjdk tests can apply to openj9 (not just hotspot), unless otherwise specified, we assume openj9 tests can be applied to hotspot unless otherwise specified.

Good point, so `AQA` is going to consume `OpenJ9` tests regardless `<impls>` restrictions unless otherwise specified.

> The reason is Java14andUp has been set only building for openj9 and ibm in build.xml. 

I think this is the actual root cause, there is no need to limit the test build to its own implementation. The test can be be built for all JVMs, and `AQA` framework will decide if it can be run.

**Comment by JasonFengJ9** [comment]
https://github.com/eclipse/openj9/pull/11146 can fix this issue (in different way). I am not going to modify the build script in this PR.

**Comment by sophia-guo** [comment]
Fix in eclipse/openj9#11146. Close this one.
